searx/engines/il_post.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Engine for Il Post, a largely independent online Italian newspaper.
+
+To use this engine add the following entry to your engines
+list in ``settings.yml``:
+
+.. code:: yaml
+
+  - name: il post
+    engine: il_post
+    shortcut: pst
+    disabled: false
+
+"""
+
+from json import loads
+from urllib.parse import urlencode
+from searx.result_types import EngineResults
+
+engine_type = 'online'
+language_support = False
+categories = ['news']
+paging = True
+page_size = 10
+
+time_range_support = True
+time_range_args = {
+    'month': 'pub_date:ultimi_30_giorni',
+    'year': 'pub_date:ultimo_anno'
+}
+
+search_api = 'https://api.ilpost.org/search/api/site_search/?'
+
+about = {
+    "website": 'https://www.ilpost.it',
+    "wikidata_id": 'Q3792882',
+    "official_api_documentation": None,
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+def request(query, params):
+    query_params = {
+        'qs': query,
+        'pg': params['pageno'],
+        'sort': "date_d",
+        'filters': 'ctype:articoli',
+    }
+
+    if params['time_range']:
```

Comment:
```suggestion
    if params['time_range']:
        if params['time_range'] not in time_range_args:
            return None
```
Since this engine doesn't support all time ranges, it shouldn't return any results for unsupported time ranges.

Reply:
thank you for pointing this out